Software/Velvet_ESP32/lib/ESP32_STM_UART/src/ESP32_STM_UART.cpp

```diff
@@ -0,0 +1,258 @@
+ * @return void
+ * @note Для корректной обработки команд необходимо, чтобы команды соответствовали протоколу общения между STM и ESP32.
+ * @brief Протокол общения между устройствами STM и ESP32
+ * @details Команды, передаваемые от STM к ESP32:
+ * * {0x01, Version0, Version1, 0x00, 0x00, 0x00, 0x00, 0x00} - CHECK_FW_CMD команда на проверку версии прошивки;
+ * * {0x02, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - BOOTLOADER_READY_CMD команда на готовность STM к загрузке новой прошивки;
+ * * {0x03, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - READY_TO_RECEIVE_DATA_CMD команда на готовность к получению данных;
+ * * {0x04, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - SEND_WEIGHT_CMD_TRANSMIT_PREPARE команда на подготовку к передаче весовых данных;
+ * * {0x05, 8B cow ID, 4B ts – unix epoch, 4B x 60 - weight float} - SEND_WEIGHT_CMD_TRANSMIT_WEIGHT_ARRAY команда на передачу весовых данных;
+ * * {0x06, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - GET_TIMESTAMP_CMD команда на получение метки времени в формате unix timestamp.
+ 
+ * Команды, передаваемые от ESP32 к STM:
+ * * {0x01, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду CHECK_FW_CMD, перезагрузить STM;
+ * * {0x01, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду CHECK_FW_CMD, ошибка соединения;
+ * *{0x01, 0x03, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду CHECK_FW_CMD, обновлений нет;
+ * * {0x02, 2B Size of FW, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду BOOTLOADER_READY_CMD, размер прошивки;
+ * * {0x03, 256B FW Data} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_PREPARE, передача данных прошивки;
+ * * {0x04, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_PREPARE, сообщает об успешной подготовке к передаче весовых данных.
+ * * {0x05, 0x01, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_WEIGHT_ARRAY, сообщает об успешной передаче весовых данных.
+ * * {0x05, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду SEND_WEIGHT_CMD_TRANSMIT_WEIGHT_ARRAY, сообщает об ошибке передачи весовых данных по WiFi.
+ * * {0x06, 4B ts – unix epoch, 0x00, 0x00, 0x00} - ответ на команду GET_TIMESTAMP_CMD, содержит запрошенную метку времени в формате unix timestamp.
+ * * {0x06, 0x02, 0X00, 0x00, 0x00, 0x00, 0x00, 0x00} - ответ на команду GET_TIMESTAMP_CMD, сообщает об ошибке при запросе метки времени.
+*/
+void ESP32_STM_UART::processIncomingMessage() {
+    if (!_serial.available()) {
+        return;
+    }
+    
+    uint8_t cmd = _serial.read();
+    
+    switch (cmd) {
+        case CHECK_FW_CMD: {
+            // Обработка CHECK_FW_CMD
+            uint8_t version0 = _serial.read();
+            uint8_t version1 = _serial.read();
```

Comment:
You HAVE TO always read all the bytes that STM sends or you'll get error when receiving next command. Ex: your code checkFirmware() was executed very fast while STM is still transmitting previous command. _serial.available() will return true and ESP will think that the next command is comming and it's 00, 00, blablabla. 
So every time you receive the command read to it's end even if it doesn't make sense.

Reply:
I need the time to think about this review.